addons/l10n_fr_hr_holidays/models/hr_leave.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+from odoo import fields, models
+
+class HrLeave(models.Model):
+    _inherit = 'hr.leave'
+
+    l10n_fr_date_to = fields.Datetime('End Date For French Rules', readonly=True)
+
+    def _compute_fr_number_of_days(self, employee, date_from, date_to, employee_calendar, company_calendar):
+        self.ensure_one()
+
+        self.l10n_fr_date_to = False
+        # We can fill the holes using the company calendar as default
+        # What we need to compute is how much we will need to push date_to in order to account for the lost days
+        # This gets even more complicated in two_weeks_calendars
+        def calendar_works_on_date(CalendarAttendance, calendar, date):
+            """
+            Returns whether the calendar has attendances planned for that day of the week
+            """
+            weektype = str(CalendarAttendance.get_week_type(date))
+            dayofweek_str = str(date.weekday())
+            return any(attendance.dayofweek == dayofweek_str and\
+                    (not calendar.two_weeks_calendar or attendance.week_type == weektype)\
+                    for attendance in calendar.attendance_ids)
```

Comment:
Looking back at this, it would probably be better performance wise to precompute the days on which the employee works, something like: (untested)
```suggestion
        working_days = defaultdict(lambda: defaultdict(lambda: False)
        for attendance in employee_calendar.attendance_ids:
            working_days[attendance.week_type][attendance.dayofweek] = True
        def calendar_works_on_date(CalendarAttendance, date):
            weektype = str(CalendarAttendance.get_week_type(date))
            dayofweek = str(date.weekday())
            return working_days[weektype][dayofweek]
```